content/en/blog/_posts/2023-04-11-kubernetes-1.27-blog.md

```diff
@@ -0,0 +1,207 @@
+
+In Kubernetes v1.25 and v1.26, this behavior toggle was part of the `SELinuxMountReadWriteOncePod` feature gate.
+
+## `JobMutableNodeSchedulingDirectives` graduates to GA
+
+This was introduced in v1.22 and started as a beta level, now it's stable. In most cases a parallel job will want the pods to run with constraints, like all in the same zone, or all either on GPU model x or y but not a mix of both. The suspend field is the first step towards achieving those semantics. Suspend allows a custom queue controller to decide when a job should start. However, once a job is unsuspended, a custom queue controller has no influence on where the pods of a job will actually land.
+
+This feature allows updating a Job's scheduling directives before it starts, which gives custom queue controllers the ability to influence pod placement while at the same time offloading actual pod-to-node assignment to kube-scheduler. This is allowed only for suspended Jobs that have never been unsuspended before. The fields in a Job's pod template that can be updated are node affinity, node selector, tolerations, labels and annotations and [scheduling gates](/docs/concepts/scheduling-eviction/pod-scheduling-readiness/). Find more details in KEP: [Allow updating scheduling directives of jobs](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/2926-job-mutable-scheduling-directives)
+
+## Mutable Pod Scheduling Directives goes to beta
+
+This allows a pod to make pod scheduling directives (nodeSelector, affinity) mutable as long as the pod is gated.  It gives the ability to mutate a pods scheduling directives before it is allowed to be scheduled, and gives an external resource controller the ability to influence pod placement while at the same time offload actual pod-to-node assignment to kube-scheduler.
+
+This opens the door for a new pattern of adding scheduling features to Kubernetes.  Specifically, building lightweight schedulers that implement features not supported by kube-scheduler, while relying on the existing kube-scheduler to support all upstream features and handle the pod-to-node binding. This pattern should be the preferred one if the custom feature doesn't require implementing a schedule plugin, which entails re-building and maintaining a custom kube-scheduler binary.
+
+## DownwardAPIHugePages graduates to stable 
+
+Support for `requests.hugepages-<pagesize>` and `limits.hugepages-<pagesize>` is being added to the downward API to be consistent with other resources like cpu, memory, and ephemeral storage. You can find more details in the KEP: [Downward API HugePages](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2053-downward-api-hugepages).
+
+# Other Updates
+
+## Graduations to stable
+
+This release includes a total of thirteen enhancements promoted to Stable:
+
+* [Mutable scheduling directives for suspended Jobs](https://github.com/kubernetes/enhancements/issues/2926)
```

Comment:
This has been covered before, so doesn't belong to "Other Updates", right?

Reply:
Not sure, As per the v1.26 release blog, they had included the enhancements mentioned in the "major themes" in the "other updates" section.

Reply:
I think we should _not_ cover major themes again under other updates. Even if a previous announcement did that, the word Other implies that we don't repeat ourselves.